### PR TITLE
Write bonding slaves config properly

### PR DESCRIPTION
**Pull request by teclator**
## Problem

When a connection config is written we are adding attributes with empty values which are irrelevant and could be omitted. In case of bonding we are also adding slaves with the _0.0.0.0_ **IPADDR** which is invalid.

- https://trello.com/c/KFilasdD/2381-3-network-ng-clean-up-configuration-to-be-written

## Solution

- Skip empty values when writing.
- Skip the write of the static _0.0.0.0_ **IPADDR**.
- Do not create empty ifroute files.

Written bonding config

### CODE REVIEW

File: src/lib/y2network/interface_config_builders/bonding.rb
Comment (imobachgs) [inline src/lib/y2network/interface_config_builders/bonding.rb:127]: I am not sure whether this logic belongs here. Please, make a note and discuss later, when we have better support for validation/errors reporting.

File: src/lib/y2network/wicked/connection_config_writers/base.rb
Comment (imobachgs) [inline src/lib/y2network/wicked/connection_config_writers/base.rb:46]: np: Perhaps we might think about adding support for such a conversion to "bootproto=" (and the rest of these methods).

**Review by imobachgs — COMMENTED** [review]
I have the impression that we should improve error handling. I am already working on something about that, so let's move forward with this PR by now.

**Comment by mchf** [comment]
Just a note ... AFAIK this was requested as a feature in the past (to store even unfilled options). There used to be even comment (!) about it in the old code.

File: src/lib/y2network/wicked/connection_config_writers/base.rb
Comment (mchf) [reply src/lib/y2network/wicked/connection_config_writers/base.rb:46]: +1 we have similar conversions (from string to a <type>) ... so it would be even nice to be consistent

File: src/lib/y2network/wicked/connection_config_writers/base.rb
Comment (mchf) [inline src/lib/y2network/wicked/connection_config_writers/base.rb:114]: this method does little bit more than described ... it is not only about checking whether it is static ip or not as method name suggests.

Comment (teclator) [reply src/lib/y2network/wicked/connection_config_writers/base.rb:114]: Well, if the bootproto is not static then it reports false as although there is an static IP it is irrelevant, and in case of static IP it should be a valid one, it is valid according to **IPAddr** but not a valid config to be written, valid_static_ip? sounds better to you? what you propose, it is a convenience method so not exposed at all.

**Review by imobachgs — APPROVED** [review]
Fine with me. Please, wait for @mchf's feedback before merging. Thanks!

File: src/lib/y2network/wicked/connection_config_writers/base.rb
Comment (mchf) [reply src/lib/y2network/wicked/connection_config_writers/base.rb:119]: yeah sounds better ... 

code should make things as clear as possible. Doesn't depend if the method is public or not. Especially now when functionality is "hidden" (nested) in a hierarchy of classes.

**Review by mchf — APPROVED** [review]
lgtm

**Pull request merged**
